mantidproject/mantid
# Add option to write different batch number for each run in various formats output by SaveReflections

### Description of work

Add option `SeparateBatchNumbers` to `SaveReflections` to write a different batch number/scale ID for each run in RunNumber column of peak table. This was requested by SXD (ISIS) who refine a separate scale for each sample rotation.

*There is no associated issue.*

**Report to:** Silvia (SXD)

### To test:
(1) Run this script
```
Load(Filename='SXD23767.raw', OutputWorkspace='SXD23767')

peaks = FindSXPeaks(InputWorkspace='SXD23767', PeakFindingStrategy='AllPeaks', AbsoluteBackground=100, ResolutionStrategy='AbsoluteResolution', XResolution=250, PhiResolution=2, TwoThetaResolution=2, OutputWorkspace='peaks')
SetUB(peaks, ub = np.array([[-0.15316025737286895, 0.3005568431977672, -0.060211365598840834], [-0.12570557266735524, -0.008794079356333284, -0.1581863032127506],[-0.23275855421286168, -0.0601758280581054, 0.25446068236675434]]))
IndexPeaks(PeaksWorkspace='peaks', Tolerance=0.1)

for ipeak,peak in enumerate(peaks):
     peak.setIntensity(ipeak)
     peak.setSigmaIntensity(np.sqrt(ipeak))
     peak.setRunNumber(2*ipeak)
    
SaveReflections(peaks, Filename=r"C:\Users\xhg73778\Desktop\test.hkl",
                SeparateBatchNumbers=True, Format="SHELX")      
```
(2) Check output file has column  (depends on format - typically `COD`) with 1:Npeaks (if SeparateBatchNumbers=False it will be a column of 1s)
(3) Test with different formats
- [ ] FullProf
- [ ] Jana



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
